### PR TITLE
clang-format 14.0.0

### PR DESCRIPTION
**Pull request by carlocab**
Created with `brew bump-formula-pr`.

`resource` blocks may require updates.

### CODE REVIEW

**Comment by carlocab** [comment]
```
  CMake Error at CMakeLists.txt:1256 (add_subdirectory):
    add_subdirectory given source
    "/tmp/clang-format-20220323-4435-1uhto3q/llvm-14.0.0.src/../third-party/benchmark"
    which is not an existing directory.
```
We probably need to adjust some more flags.

File: Formula/clang-format.rb [hunk 1]
```diff
@@ -47,31 +46,43 @@ def install
...
     # NB: below C code is messily formatted on purpose.
     (testpath/"test.c").write <<~EOS
       int         main(char *args) { \n   \t printf("hello"); }
     EOS
+    system "git", "add", "test.c"
 
     assert_equal "int main(char *args) { printf(\"hello\"); }\n",
         shell_output("#{bin}/clang-format -style=Google test.c")
+
+    ENV.prepend_path "PATH", bin
```
Comment (carlocab) [inline Formula/clang-format.rb:85]: A better solution might be to wrap `git-clang-format` in an env script, maybe.

File: Formula/clang-format.rb [hunk 2]
```diff
@@ -47,31 +46,44 @@ def install
...
 
     mkdir llvmpath/"build" do
       args = std_cmake_args
-      args << "-DLLVM_EXTERNAL_PROJECTS=\"clang\""
+      args << "-DLLVM_EXTERNAL_PROJECTS=clang"
+      args << "-DLLVM_INCLUDE_BENCHMARKS=OFF"
       args << ".."
-      system "cmake", "-G", "Ninja", *args
+      system "cmake", *args
       system "ninja", "clang-format"
```
Comment (carlocab) [inline Formula/clang-format.rb:60]: Oops. This should be `make`. Will fix it when I get back to my computer.

File: Formula/clang-format.rb [hunk 3]
```diff
@@ -47,31 +46,44 @@ def install
...
     mkdir llvmpath/"build" do
       args = std_cmake_args
-      args << "-DLLVM_EXTERNAL_PROJECTS=\"clang\""
+      args << "-DLLVM_EXTERNAL_PROJECTS=clang"
+      args << "-DLLVM_INCLUDE_BENCHMARKS=OFF"
       args << ".."
-      system "cmake", "-G", "Ninja", *args
-      system "ninja", "clang-format"
+      system "cmake", *args
+      system "make", "clang-format"
```
Comment (gjasny) [inline Formula/clang-format.rb:62]: Better use `"cmake", "--build", ".", "--target", "clang-format"` That's build system agnostic.

File: Formula/clang-format.rb [hunk 4]
```diff
@@ -32,7 +32,6 @@ class ClangFormat < Formula
   end
 
   depends_on "cmake" => :build
-  depends_on "ninja" => :build
```
Comment (gjasny) [inline Formula/clang-format.rb:35]: Is there a reason why you dropped ninja? When building, ninja parallelizes by default. Make does not.

Comment (carlocab) [reply Formula/clang-format.rb:35]: We set `MAKEFLAGS` to parallelise `make` builds. The extra dependency shouldn't be needed.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**